src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1359,7 +1359,10 @@ private bool ShouldCache(long number)
                 return null;
             }
 
-            Block block = _blockStore.Get(blockHash, shouldCache: false);
+            BlockHeader header = FindHeader(blockHash, options);
```

Comment:
Could we get just a number from the hash without getting whole header when combined with the second PR that prepends headers with block number and adds a second index?

Reply:
Yea, thats another reason for the other PR.